brainglobe_atlasapi/atlas_generation/atlas_scripts/template_script.py

```diff
@@ -0,0 +1,111 @@
+
+    If possible, please use the Pooch library to retrieve any resources.
+    """
+    pass
+
+
+def retrieve_template_and_reference():
+    """
+    Retrieve the desired template and reference as two numpy arrays.
+
+    Returns:
+        tuple: A tuple containing two numpy arrays. The first array is the template volume, and the second array is the reference volume.
+    """
+    template = None
+    reference = None
+    return template, reference
+
+
+def retrieve_hemisphere_map():
+    """
+    Retrieve a hemisphere map for the atlas.
+
+    If your atlas is asymmetrical, you may want to use a hemisphere map. This is an array in the same shape as your template,
+    with 0's marking the left hemisphere, and 1's marking the right.
+
+    If your atlas is symmetrical, ignore this function.
+
+    Returns:
+        numpy.array or None: A numpy array representing the hemisphere map, or None if the atlas is symmetrical.
+    """
+    return None
+
+
+def retrieve_structure_information():
+    """
+    This function should return a pandas DataFrame with information about your atlas.
+
+    The DataFrame should be in the following format:
+
+    ╭─────────────┬───────────────────────────────────┬─────────┬───────────────────────┬───────────────────╮
+    | id          | name                              | acronym | structure_id_path     | rgb_triplet       |
+    |             |                                   |         |                       |                   |
+    ├─────────────┼───────────────────────────────────┼─────────┼───────────────────────┼───────────────────┤
+    | 997         | root                              | root    | []                    | [255, 255, 255]   |
+    ├─────────────┼───────────────────────────────────┼─────────┼───────────────────────┼───────────────────┤
+    | 8           | Basic cell groups and regions     | grey    | [997]                 | [191, 218, 227]   |
+    ├─────────────┼───────────────────────────────────┼─────────┼───────────────────────┼───────────────────┤
+    | 567         | Cerebrum                          | CH      | [997, 8]              | [176, 240, 255]   |
+    ╰─────────────┴───────────────────────────────────┴─────────┴───────────────────────┴───────────────────╯
+
+    Returns:
+        pandas.DataFrame: A DataFrame containing the atlas information.
+    """
+    return None
+
+
+def retrieve_or_construct_meshes():
+    """
+    This function should return a dictionary of ids and corresponding paths to mesh files.
+    Some atlases are packaged with mesh files, in these cases we should use these files.
+    Then this function should download those meshes. In other cases we need to construct
+    the meshes ourselves. For this we have helper functions to achieve this.
+    """
+    meshes_dict = {}
+    return meshes_dict
+
+
+### If the code above this line has been filled correctly, nothing needs to be edited below (unless variables need to be passed between the functions).
```

Comment:
Belated thought. This (and its equivalent in the example script) should be inside `if __name__ == "__main__" guards` to avoid atlas creation code being executed when this module is imported.

Reply:
sure thing, i hadn't considered that